x-pack/plugins/licensing/common/license.ts

```diff
@@ -91,9 +99,11 @@ export class License implements ILicense {
       return this._signature;
     }
 
-    this._signature = createHash('md5')
-      .update(JSON.stringify(this.toObject()))
-      .digest('hex');
+    if (!this.plugin.sign) {
```

Comment:
It looks like both plugins have this method. When is this case hit?

Also maybe instead of passing the entire plugin to this class, we should just provide a `sign` function.

Reply:
This reason passing the plugin for a single function didn't make sense is because I also forgot to expose refreshing on the license itself. I will revise.